hack/test-stirunimage.sh

```diff
@@ -14,6 +14,10 @@ function time_now()
 
 mkdir -p /tmp/sti
 WORK_DIR=$(mktemp -d /tmp/sti/test-work.XXXX)
+S2I_WORK_DIR=${WORK_DIR}
+if [[ "$OSTYPE" == "cygwin" ]]; then
```

Comment:
Cygwin becomes a prerequisite for using our build and test scaffolding on Windows, but not for running the s2i binary itself, which should work on cygwin and non-cygwin environments (although there is no special testing for the latter currently).

Those who want to build on Windows without cygwin can run go build just as they could before.

I think this is the best and most realistic approach here.